sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -294,17 +294,22 @@ private[sql] abstract class BaseWriterContainer(
 
   private def newOutputCommitter(context: TaskAttemptContext): OutputCommitter = {
     val committerClass = context.getConfiguration.getClass(
-      "mapred.output.committer.class", null, classOf[OutputCommitter])
+      "mapreduce.output.committer.class", null, classOf[OutputCommitter])
 
     Option(committerClass).map { clazz =>
-      val ctor = clazz.getDeclaredConstructor(classOf[Path], classOf[TaskAttemptContext])
-      ctor.newInstance(new Path(outputPath), context)
+      if (classOf[FileOutputCommitter].isAssignableFrom(clazz)) {
```

Comment:
Would be better to have an explicit import alias `MapReduceFileOutputCommitter` for this.